### DISCUSSION TITLE
HAVING ISSUES WITH UNIT AND INT

### DISCUSSION THREAD

**Discussion by FetemFT** [Cyfrin/foundry-full-course-cu#2606] [General]
Just when i get to unit, talking about favoritenumber, i followed what patrick say and i'm still getting error

**Comment by EngrPips** [comment]
Hello @FetemFT, What error are you getting?

**Comment by FetemFT** [comment]
![WhatsApp Image 2024-10-03 at 12 32 57_27337f49](https://github.com/user-attachments/assets/f40ff54b-cca3-4b86-8df1-7d4a9dd21087)

**Reply by FetemFT** [reply]
I cannot figure out the error there, tried everything

**Reply by EngrPips** [reply]
Hello @FetemFT, at line `7` of your code you need to end with `;`

**Reply by FetemFT** [reply]
I just did that and the error moves to line 8

**Reply by EngrPips** [reply]
Hello @FetemFT, are you not following the course?

**Reply by FetemFT** [reply]
I'm following just got lost on the way, will rewatch

**Reply by EngrPips** [reply]
Yeah, Whenever you get lost, rewind so that you get back on track. Maybe take a break if the content is getting too much.

**Reply by FetemFT** [reply]
Thanks

**Reply by EngrPips** [reply]
You are welcome, buddy.

**Comment by felixoder** [comment]
@FetemFT that is not unit that should be uint [unsigned integer] by default uint256 but uint is ok
and also the store function is not technically correct 

the line should be 

```
string favouriteNumber = '88'; // a termination semicolon

function store(uint256 _favouriteNumber) public {
        myFavouriteNumber = _favouriteNumber;
    }
```
There you go

**Reply by FetemFT** [reply]
Thanks. I will try and rewrite it cause it doesn't seem to work; one question I thought we have to put in the values of strings in a colon?
